cypress/e2e/ui/Compute/Clouds/Providers/cloud_provider.cy.js

```diff
@@ -0,0 +1,4903 @@
+          flashClassMap.success,
+          FLASH_MESSAGE_OPERATION_CANCELLED
+        );
+      });
+
+      it('Verify successful validate + add/refresh/delete operations', () => {
+        const nameFieldValue = `${VMWARE_VCLOUD_NAME_VALUE} - Verify validate/add/refresh/delete operations`;
+        // Add all mandatory field values
+        fillVmwareVcloudForm({
+          nameValue: nameFieldValue,
+          hostValue:
+            'vmware-vcloud.verify-validate-add-refresh-delete-operations.com',
+        });
+        // Assert validation success message
+        validate({
+          stubErrorResponse: false,
+        });
+        assertValidationSuccessMessage();
+        // Assert add button is now enabled and then perform add operation
+        cy.getFormFooterButtonByTypeWithText({
+          buttonText: 'Add',
+          buttonType: 'submit',
+        })
+          .should('be.enabled')
+          .click();
+        // Assert add operation success flash message
+        cy.expect_flash(flashClassMap.success, FLASH_MESSAGE_OPERATION_SAVED);
+        // Select the created provider from the data table
+        selectCreatedProvider(nameFieldValue);
+        // Select refresh option and assert flash message
+        cy.expect_browser_confirm_with_text({
+          confirmTriggerFn: () =>
+            cy.toolbar(CONFIG_TOOLBAR_BUTTON, REFRESH_CONFIG_OPTION),
+          containsText: REFRESH_OPERATION_MESSAGE,
+        });
+        cy.expect_flash(flashClassMap.success, REFRESH_OPERATION_MESSAGE);
+
+        // FIXME: remove this block once bug is fixed
+        // Bug: After refresh, config option other than add remains disabled and requires any action to be performed to enable it back
+        /* ==================================================================== */
+        cy.toolbar(CONFIG_TOOLBAR_BUTTON, ADD_PROVIDER_CONFIG_OPTION);
+        cy.getFormFooterButtonByTypeWithText({ buttonText: 'Cancel' }).click();
+        /* ==================================================================== */
+
+        // Delete the created provider & assert delete operation success flash message
+        selectProviderAndDeleteWithOptionalFlashMessageCheck({
+          createdProviderName: nameFieldValue,
+          assertDeleteFlashMessage: true,
+        });
+      });
+    });
+
+    describe('Validate edit form', () => {
+      /**
+       * The provider name is set in this variable at the start of each test,
+       * allowing afterEach to identify it for deletion.
+       */
+      let nameFieldValue;
+
+      it('Validate visibility of elements', () => {
+        nameFieldValue = `${VMWARE_VCLOUD_NAME_VALUE} - Verify edit form elements`;
+        // Setting up data and opening edit form
+        addVmwareVcloudProviderAndOpenEditForm({
+          nameValue: nameFieldValue,
+          hostValue: 'vmware-vcloud.verify-edit-form-elements.com',
+        });
+
+        // Validate elements
+        validateVmwareVcloudFormFields(true);
```

Comment:
So, I can't review this whole thing because it's huge but I think creating shared helper functions could help eliminate some duplication and dry up some of the extra code here.  For example, this function is defined in each of the cloud provider vendor types.  A shared function should be the next step.  

Maybe let's get this in and then refactor it into shared functions.  If we're still at several hundreds of lines, maybe we need to split out the tests per vendor.